main.go

```diff
@@ -32,3 +40,65 @@ func main() {
 		panic(err.Error())
 	}
 }
+
+func getIPAMClient(ctx context.Context, root settings.Source) (*ipamfetcher.Client, error) {
+	ipsPrefixedEnv := &settings.PrefixSource{
+		Source: root,
+		Prefix: []string{"IPS"},
+	}
+	subnetsPrefixedEnv := &settings.PrefixSource{
+		Source: root,
+		Prefix: []string{"SUBNETS"},
+	}
+	customersPrefixedEnv := &settings.PrefixSource{
+		Source: root,
+		Prefix: []string{"CUSTOMERS"},
+	}
+	device42IPsClientComponent := &ipamfetcher.Device42ClientComponent{}
+	device42IPsConfig := new(ipamfetcher.Device42ClientConfig)
+	if err := settings.NewComponent(ctx, ipsPrefixedEnv, device42IPsClientComponent, device42IPsConfig); err != nil {
+		return nil, err
+	}
+	devicesFetcher := &ipamfetcher.Device42DeviceFetcher{
+		Iterator: &ipamfetcher.Device42PageIterator{
+			Context: ctx,
```

Comment:
Question: I must have missed this in the previous PR. Why does the iterator take a context? Won't this just be context.Background() from main?

Reply:
this function accepts a `context.Context`, which is `context.Background()` from `main` - the iterator then passes this along to the `PageFetcher` which actually makes outbound requests

Reply:
From our offline conversation: the context shouldn't be passed in this way. I think we agreed the iterator should be created on each invocation with the relevant context at that point in time. Perhaps the `Device42DeviceFetcher` could take an Iterator generator instead.

Reply:
This might be out of scope for this PR, if you just want to create a ticket for fixing it.